core/src/main/resources/hudson/PluginManager/advanced.jelly

```diff
@@ -63,13 +63,31 @@ THE SOFTWARE.
                             <f:block>
                                 <!-- @size is for other browsers, @style is for IE -->
                                 ${%File}:
-                                <input type="file" name="name" class="setting-input" style="width:80%" size='40'/>
+                                <input type="file" name="name" class="setting-input" style="width:80%" size='40' accept=".hpi,.jpi"/>
                             </f:block>
                             <f:block>
                                 <f:submit value="${%Upload}"/>
                             </f:block>
                         </f:form>
                     </section>
+                    <section>
+                        <h2>${%Deploy from URL}</h2>
```

Comment:
I made some small modifications to the UI, I don't think they address you concerns about the buttons but they simplify a little bit

![image](https://user-images.githubusercontent.com/55871/139507973-6a705b42-9bce-4950-bb8a-1c954bdff61d.png)

Reply:
> the UX here isn't the best imo.
> 
> I think either:
> 
> * an expandable radio section where you select what type and a field pops out
> * just a radio option and a single field, but depends on messaging.
> 
> radio options would be:
> 
> * File
> * URL
> 
> the main issue I think is soo many primary buttons so close together:
> 
> ![image](https://user-images.githubusercontent.com/21194782/139422889-09c3829d-34fc-4942-b461-4e77a366e4fa.png)
> 
> and also these are the same thing even is the source is different so I don't think it justifies another form.
> 
> any thoughts @fqueiruga @janfaracik @uhafner?

Yes, from a user experience view it would make more sense to have one "Deploy" button with two fields. One for the file and one for the URL. I don't think that an additional radio button is required. 

(But this view and the upload concept is also in a bad UI shape yet so it will not make it dramatically difference)

Reply:
I had it originally as just two fields but @daniel-beck asked me to separate into two sections. Perhaps it could be combined into one section and the text "Upload Plugin" could be changed to "Deploy Plugin" with the two fields?

Reply:
Yes, that would be better. I wonder if it would be even better if we would add an additional tab for the upload. So we would have 
- Updates
- Available
- Installed
- Deploy
- Advanced (or better Settings?)

Then we can use the whole form with a single primary button for the upload.

Reply:
Here is what a combined section would look like:

![image](https://user-images.githubusercontent.com/55871/139593017-92267fd5-2f12-4920-bc8e-af93b8a4eb3c.png)

Reply:
Here is what a combined section would look like:

![image](https://user-images.githubusercontent.com/55871/139593054-77adcc2e-bfb1-4078-ad7a-db598ed8905e.png)

Reply:
The text under deploy plugin needs changing from upload, I would change it to say plugin instead of `.hpi`.

between the file and url fields what about adding a `<p>Or</p>`

Reply:
Something like this? 
![image](https://user-images.githubusercontent.com/55871/139604592-9bd1db6b-a896-4441-841d-f73c1fc56fef.png)

Reply:
Much nicer, not sure if those - are needed

Reply:
(Personal opinion) I really prefer the approach with the "or" instead of having to click on a radio to select the type of input. This adds a non-necessary click I would say :)

Reply:
Yes, this looks good!

Reply:
@timja I wasn't sure about the - either, but have seen it in other places on the web. Here is the current commit: 
![image](https://user-images.githubusercontent.com/55871/139706226-a22a1126-e9e7-4e3c-bd13-b3a537394925.png)

Reply:
@timja I wasn't sure about the - either, I had seen it in other places on the web, but I think it looks better without it. Here is what it looks like with the current commit:

![image](https://user-images.githubusercontent.com/55871/139706425-999e23ae-7b69-4d46-9749-449e001eecbf.png)